low-level/files/opt/seagate/sspl/setup/cortx_manifest/lr2/manifest_storage.py

```diff
@@ -0,0 +1,381 @@
+            "enclosures": self.get_enclosures_info,
+            "controllers": self.get_controllers_info,
+            "power-supplies": self.get_psu_info,
+            "fan-modules": self.get_fan_modules_info,
+            "drives": self.get_drives_info,
+            "sideplane": self.get_sideplane_expander_info
+        }
+
+    def validate_storage_type_support(self):
+        """Check for supported storage type."""
+        storage_type = Conf.get(GLOBAL_CONF, STORAGE_TYPE_KEY, "virtual").lower()
+        logger.debug(self.log.svc_log(
+            f"Storage Type:{storage_type}"))
+        supported_types = ["5u84", "rbod", "pods", "corvault"]
+        if storage_type not in supported_types:
+            msg = f"Manifest provider is not supported for storage type {storage_type}"
+            logger.error(self.log.svc_log(msg))
+            raise ManifestError(errno.EINVAL, msg)
+
+    def get_manifest_info(self, rpath):
+        """
+        Fetch Manifest information for given rpath
+
+        rpath: Manifest path to fetch its information
+               Examples:
+                    node>storage[0]
+                    node>storage[0]>controllers
+        """
+        logger.info(self.log.svc_log(
+            f"Get Manifest data for rpath:{rpath}"))
+        info = {}
+        resource_found = False
+        nodes = rpath.strip().split(">")
+        leaf_node, _ = self.get_node_details(nodes[-1])
+
+        # Fetch manifest information for all sub nodes
+        if leaf_node == "storage":
+            resource_found = True
+            info = self.get_storage_manifest_info()
+        elif leaf_node in self.storage_frus:
+            resource_found = True
+            info = {leaf_node: self.storage_frus[leaf_node]()}
+
+        if not resource_found:
+            msg = f"Invalid rpath or manifest provider doesn't have support for'{rpath}'."
+            logger.error(self.log.svc_log(msg))
+            raise ManifestError(
+                errno.EINVAL, msg)
+        return info
+
+    def get_storage_manifest_info(self):
+        """Get storage enclosure information"""
+        storage = []
+        info = {}
+        for fru, method in self.storage_frus.items():
+            try:
+                info.update({fru: method()})
+            except Exception as err:
+                msg = f"Unable to get data for '{fru}' resource.: {err}"
+                logger.error(self.log.svc_log(msg))
+                info.update({fru: None})
+        storage.append(info)
+        return storage
+
+    def get_enclosures_info(self):
+        """Update and return enclosure information in specific format"""
+        data = []
+        chassis_serial_no = "NA"
+        enclosure = self.get_realstor_encl_data("enclosures")
+        frus = self.get_realstor_encl_data("frus")
+        if frus:
+            for fru in frus:
+                if fru["name"] == "CHASSIS_MIDPLANE":
+                    chassis_serial_no = fru.get("serial-number", "NA")
```

Comment:
Keep `break` inside the condition. That will avoid unnecessary iteration.

Reply:
done